src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelEventGenerator.java

```diff
@@ -51,9 +63,46 @@ protected static String createRepoURI(String url, String projectName) {
         }
     }
 
-    private static String getSshBaseUrl(String host) throws URISyntaxException {
+    protected static String getPreviousEiffelEvent(final String linkedEiffelEventType, final String projectName,
+            final String searchCriteria, final File pluginDirectoryPath) {
+        try {
+            EventStorage eventStorage = EventStorageFactory.getEventStorage(pluginDirectoryPath, linkedEiffelEventType);
+            String lastEiffelEvent = getEiffelEventIdFromStorage(eventStorage, projectName, searchCriteria);
+            return lastEiffelEvent;
+        } catch(IllegalArgumentException e) {
+            LOGGER.error("Could not get previous Eiffel event.", e);
+            return "";
+        }
+    }
+
+    protected static Link createLink(final String linkType, final String lastEiffelEvent) {
+        if (!StringUtils.isEmpty(lastEiffelEvent)) {
+            Link link = new Link();
+            link.type = linkType;
+            link.target = lastEiffelEvent;
+
+            return link;
+        }
+        return null;
+    }
+
+    private static String getSshBaseUrl(final String host) throws URISyntaxException {
         URI uri;
         uri = new URI("ssh", null, host, DEFAULT_SSH_PORT, "/", null, null);
         return uri.toString();
     }
+
+    private static String getEiffelEventIdFromStorage(final EventStorage eventStorage, final String projectName,
+            final String searchCriteria) {
+        try {
+            String eventId = eventStorage.getEventId(projectName, searchCriteria);
```

Comment:
may be return without having to add to a string something like
return eventStorage.getEventId(...)

Reply:
I think that's less readable. Better adding to string before returning